smtp-server/smtp-server.d.ts

```diff
@@ -0,0 +1,92 @@
+// Type definitions for smtp-server v1.15.0
+// Project: https://github.com/andris9/smtp-server
+// Definitions by: markisme <https://github.com/markisme/>
+// Definitions: https://github.com/markisme/smtp-server.git
+
+import { Stream } from 'typed-stream';
+
+export class SMTPServer {
+    constructor(options: SMTPServerOptions);
+    close(callback: (err?: Error) => any): any;
+    listen(...args: any[]): void;
+    onAuth(auth: Authentication, session: Session, callback: any): any;
+    onClose(callback: (err?: Error) => any): any;
+    onConnect(session: Session, callback: (err?: Error) => any): any;
+    onData(stream: Stream, session: Session, callback: (err?: Error) => any): any;
+    onMailFrom(address: Address, session: Session, callback: (err?: Error) => any): any;
+    onRcptTo(address: Address, session: Session, callback: (err?: Error) => any): any;
+}
+
+export interface SMTPServerOptions {
+    secure?: boolean; //default is false.
+    key?: Buffer;
+    cert?: Buffer;
+    ca?: Buffer;
+    name?: string;//optional hostname of the server.
```

Comment:
Why not use a JSDoc comment for this? `/** Optional hostname of the server. */` on a line above it. JSDoc comments are used for an editor to provide info on hover, while this comment will be ignored.